compiler/rustc_session/src/config.rs

```diff
@@ -1005,6 +1006,92 @@ pub fn to_crate_config(cfg: FxHashSet<(String, Option<String>)>) -> CrateConfig
     cfg.into_iter().map(|(a, b)| (Symbol::intern(&a), b.map(|b| Symbol::intern(&b)))).collect()
 }
 
+/// The parsed `--check-cfg` options
+pub struct CheckCfg<T = String> {
+    /// Set if `names()` checking is enabled
+    pub names_checked: bool,
+    /// The union of all `names()`
+    pub names_valid: FxHashSet<T>,
+    /// The set of names for which `values()` was used
+    pub values_checked: FxHashSet<T>,
+    /// The set of all (name, value) pairs passed in `values()`
+    pub values_valid: FxHashSet<(T, T)>,
+}
+
+impl<T> Default for CheckCfg<T> {
+    fn default() -> Self {
+        CheckCfg {
+            names_checked: false,
+            names_valid: FxHashSet::default(),
+            values_checked: FxHashSet::default(),
+            values_valid: FxHashSet::default(),
+        }
+    }
+}
+
+/// Converts the crate `--check-cfg` options from `String` to `Symbol`.
+/// `rustc_interface::interface::Config` accepts this in the compiler configuration,
+/// but the symbol interner is not yet set up then, so we must convert it later.
+pub fn to_check_config(cfg: CheckCfg) -> CrateCheckConfig {
+    let mut res = CrateCheckConfig {
+        names_checked: cfg.names_checked,
+        names_valid: cfg.names_valid.into_iter().map(|a| Symbol::intern(&a)).collect(),
+        values_checked: cfg.values_checked.into_iter().map(|a| Symbol::intern(&a)).collect(),
+        values_valid: cfg
+            .values_valid
+            .into_iter()
+            .map(|(a, b)| (Symbol::intern(&a), Symbol::intern(&b)))
+            .collect(),
+    };
+    res.names_valid.extend(res.values_checked.iter().copied());
```

Comment:
This operation is not a part of the `CheckCfg` -> `CrateCheckConfig` conversion, it's some separate logic.
Maybe it should be moved to the end of `fn parse_check_cfg`.